python/tester/upload.py

```diff
@@ -391,6 +383,15 @@ def upload_from_pool(visit, snap_id):
                         md[header_key] = headers[header_key]
                     dest_bucket.put_object(Body=json.dumps(md), Key=filename_sidecar)
 
+            # r+b required by replace_header_key.
+            with tempfile.TemporaryFile(mode="r+b") as buffer:
+                src_bucket.download_fileobj(src_blob.key, buffer)
+                for header_key in headers:
+                    replace_header_key(buffer, header_key, headers[header_key])
+                buffer.seek(0)  # Assumed by upload_fileobj.
+                dest_bucket.upload_fileobj(buffer, filename)
+            _log.debug(f"{filename} is uploaded to {dest_bucket}")
+
```

Comment:
Merge with "Upload json files..."?

Reply:
Doing this in `upload_from_repo.py` is now in its one commit (https://github.com/lsst-dm/prompt_processing/pull/154/commits/aa13d283c020f86cfb66e89b8c9a1b174656c1fd) 

Doing this for `upload.py` is independent of this ticket and is still in a separate commit (https://github.com/lsst-dm/prompt_processing/pull/154/commits/ddcb9d3a9013b4d8dc09ed5bce4cf679295470e2)